src/main/java/com/kryptnostic/rhizome/emails/EmailStreamSerializer.java

```diff
@@ -0,0 +1,125 @@
+package com.kryptnostic.rhizome.emails;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+
+import jersey.repackaged.com.google.common.collect.Lists;
+import jodd.mail.Email;
+import jodd.mail.EmailMessage;
+import jodd.mail.MailAddress;
+
+import com.geekbeast.rhizome.pods.hazelcast.SelfRegisteringStreamSerializer;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+
+public class EmailStreamSerializer implements SelfRegisteringStreamSerializer<Email> {
+
+    @Override
+    public void write( ObjectDataOutput out, Email object ) throws IOException {
+        serialize( out, object );
+    }
+
+    @Override
+    public Email read( ObjectDataInput in ) throws IOException {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public int getTypeId() {
+        return 0;
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public Class<Email> getClazz() {
+        return Email.class;
+    }
+
+    public static void serialize( ObjectDataOutput out, Email object ) throws IOException {
+        serializeMailAddresses( out, object.getFrom() );
+        serializeMailAddresses( out, object.getTo() );
+        serializeMailAddresses( out, object.getCc() );
+        serializeMailAddresses( out, object.getBcc() );
+        serializeMailAddresses( out, object.getReplyTo() );
+        out.writeUTF( object.getSubject() );
+        out.writeUTF( object.getSubjectEncoding() );
+        out.writeInt( object.getPriority() );
+        out.writeLong( object.getSentDate().getTime() );
```

Comment:
Though really we should just use findbugs and ```@CheckForNull``` and ```@Nonnull``` to guarantee that we need to or don't need to null check this